terraform/deployments/elasticsearch/iam.tf

```diff
@@ -58,6 +58,6 @@ data "aws_iam_policy_document" "can_configure_es_snapshots" {
   }
   statement {
     actions   = ["es:ESHttpPut"]
-    resources = ["${aws_elasticsearch_domain.opensearch.arn}/*"]
+    resources = [for d in aws_elasticsearch_domain.opensearch : "${d.arn}/*"]
```

Comment:
Sadly I'm 99% certain this will fail on apply, terraform doesn't validate it, but IAM policy statements which don't reference any resources aren't valid and when the count is 0 that will make this an empty array.

You'll have to use a [dynamic block](https://developer.hashicorp.com/terraform/language/expressions/dynamic-blocks) for this statement so the whole statement doesn't get created